### ISSUE TITLE
Impossible to use PrestaSharp with new Xamarin.Forms project...

### ISSUE THREAD

**Issue by espritm** [Bukimedia/PrestaSharp#285]
Hello everyone,

I would like to use PrestaSharp to create a Xamarin.Forms Android and iOS mobile app.
Unfortunately, I encounter many problems and it is impossible for me to contact my prestashop website with PrestaSharp. Could you help me please ?

Here are the steps to reproduce from scratch : 

- Run Visual Studio and create a new "Cross Platform App (Xamarin.Forms or Native)" project.

- Choose Blank App, Xamarin.Forms, and Portable Class Library (PCL)

- Open the solution's Nuget Package Manager, and try to install Bukimedia.PrestaSharp on Android and iOS projects.

- Get an error message : 
`"Could not install package 'RestSharp 105.0.1'. You are trying to install this package into a project that targets 'MonoAndroid,Version=v7.1', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author."`

- PrestaSharp depends on RestSharp. So the Nuget Package Manager, search for RestSharp, and install latest version (actually 105.2.3) on Android and iOS projects.

- Compile and deploy your Android app => OK, no problem

- Now install PrestaSharp on Android and iOS project (actually version 1.157.0)

- Compile and deploy your Android app => it does not compile. You get the following error message : 
```
"Exception while loading assemblies: System.IO.FileNotFoundException: Could not load assembly 'System.Configuration, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Perhaps it doesn't exist in the Mono for Android profile?
Nom de fichier : 'System.Configuration.dll'
   à Java.Interop.Tools.Cecil.DirectoryAssemblyResolver.Resolve(AssemblyNameReference reference, ReaderParameters parameters)
   à Xamarin.Android.Tasks.ResolveAssemblies.AddAssemblyReferences(DirectoryAssemblyResolver resolver, ICollection`1 assemblies, AssemblyDefinition assembly, Boolean topLevel)
   à Xamarin.Android.Tasks.ResolveAssemblies.Execute(DirectoryAssemblyResolver resolver)	EpicuriaSucreSale.Android	C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets"
```

 - On your Android project, add a reference to "C:\Windows\Microsoft.NET\Framework64\v4.0.30319\System.Configuration.dll". Doing that is really bad for production, but it compiles, is enough for a proof of concept, and I didn't find another solution.

- Add a button to your Page, and when the button is clicked, call this code : 
```
            CartFactory cartFactory = new CartFactory(sYourPrestashopURL, sYourPrestashopAPIKey, sYourPrestashopAPIPassword);
            var tmp = cartFactory.Get(iID);
```

- When this code is executed, cartFactory.Get(iID); throw the following exception : 
```
System.TypeLoadException: Could not resolve type with token 01000056
   at Bukimedia.PrestaSharp.Factories.CartFactory.Get (System.Int64 CartId) [0x00017] in <f1ca8b22b2584ce08710daa6cf5997d0>:0 
   at PrestaPOC.Droid.ButtonPresta.GetCartFromID (System.Int32 iID) [0x00016] in C:\Workspace\PrestaPOC\PrestaPOC\PrestaPOC.Android\ButtonPresta.cs:15 
   at PrestaPOC.MainPage.BtnPresta_Clicked (System.Object sender, System.EventArgs e) [0x00001] in C:\Workspace\PrestaPOC\PrestaPOC\PrestaPOC\MainPage.xaml.cs:21 
   at Xamarin.Forms.Button.Xamarin.Forms.IButtonController.SendClicked () [0x00020] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Core\Button.cs:125 
   at Xamarin.Forms.Platform.Android.AppCompat.ButtonRenderer+ButtonClickListener.OnClick (Android.Views.View v) [0x0000b] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\AppCompat\ButtonRenderer.cs:291 
   at Android.Views.View+IOnClickListenerInvoker.n_OnClick_Landroid_view_View_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_v) [0x00011] in /Users/builder/data/lanes/4009/9578cdcd/source/monodroid/src/Mono.Android/platforms/android-25/src/generated/Android.Views.View.cs:1857 
   at (wrapper dynamic-method) System.Object:b3822dae-4195-4828-9258-0b3725f17771 (intptr,intptr,intptr)
```

- I googled for that and the exception seems to be throwed because there is two references to a dll but of different version. I am pretty sure it's because of System.Configuration.dll
https://stackoverflow.com/a/42511504/4965913

And here I am, after two entire days working on that POC, I can't get carts from my prestashop website.
Please I need your help :)  What should I do to get it working ? What did I do wrong ?

I put the project on Github so you can get it to reproduce the exception : https://github.com/espritm/PrestaPOC

Thanks a lot for any help 👍

**Comment by espritm** [comment]
Hello everyone,

After few researches and some help (thanks to [stackoverflow](https://stackoverflow.com/q/46399981/4965913)), the problem seems to be because of Restsharp.

If the project PrestaSharp is compiled with the latest version of Restsharp, actualy v105.2.3, everything is ok.

Is there any chance to get the nuget package PrestaSharp updated with the latest version of RestSharp ?

Thanks in advance :)

**Comment by mowcixo** [comment]
Hello @espritm, sorry for the delay.

Initially PrestaSharp was only flavored for .NET Framework so the use with Xamarin was not supported. Since **v1.0.2** PrestaSharp supports .NET Standard, so it should work with Xamarin, we didn't tested though.

It would be great if you test it now and let us know.

**Comment by espritm** [comment]
Hey @mowcixo  !
Thank you for your answer ! 

I tested this morning and yes, it's working as a charm ! :)

I have created a Xamarin Forms project with .NET Standard 2.0 (instead of PCL). This is the new version of Xamarin.

I installed the nuget PrestaSharp V1.0.3 (latest) on all solution's projects, and it compiles well.
I instanciated a CartFactory and get a Cart from an ID, and again it is working well.

Thank you 👍
